File: examples/java/struct-def/src/main/java/io/littlehorse/examples/ParkingTicketReport.java
package io.littlehorse.examples;

import io.littlehorse.sdk.worker.LHStructDef;
import io.littlehorse.sdk.worker.LHStructField;

@LHStructDef(
        value = "parking-ticket-report",
        description =
                "A parking ticket report capturing vehicle details and license plate at the time of the violation.")
public record ParkingTicketReport(
        @LHStructField(description = "The vehicle's manufacturer or brand name, ex: Suzuki") String vehicleMake,
        @LHStructField(description = "The specific model and iteration of the vehicle, ex: Swift") String vehicleModel,
        @LHStructField(
                        description =
                                "The license plate number for the vehicle at the time of writing the report, ex: C90-ELE")
                String licensePlateNumber) {
    @Override
    public String toString() {
        return String.format("%s %s, Plate Number: %s", vehicleMake, vehicleModel, licensePlateNumber);
    }
}

File: examples/dotnet/StructDefExample/ParkingTicketReport.cs
using LittleHorse.Sdk.Worker;

namespace StructDefExample;

[LHStructDef("parking-ticket-report", "A parking ticket report capturing vehicle details and license plate at the time of the violation.")]
public class ParkingTicketReport
{
    [LHStructField(description: "The vehicle's manufacturer or brand name, ex: Suzuki")]
    public string VehicleMake { get; set; } = string.Empty;

    [LHStructField(description: "The specific model and iteration of the vehicle, ex: Swift")]
    public string VehicleModel { get; set; } = string.Empty;

    [LHStructField(description: "The license plate number for the vehicle at the time of writing the report, ex: C90-ELE")]
    public string LicensePlateNumber { get; set; } = string.Empty;

    public ParkingTicketReport()
    {
    }

    public ParkingTicketReport(string vehicleMake, string vehicleModel, string licensePlateNumber)
    {
        VehicleMake = vehicleMake;
        VehicleModel = vehicleModel;
        LicensePlateNumber = licensePlateNumber;
    }

    public override string ToString()
    {
        return $"{VehicleMake} {VehicleModel}, Plate Number: {LicensePlateNumber}";
    }
}
